packages/block-editor/src/components/roving-tab-index/container.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { RovingTabIndexProvider } from './context';
+
+function RovingTabIndexContainer( { children, ...props } ) {
+	const [ lastFocusedElement, setLastFocusedElement ] = useState();
+
+	return (
+		<RovingTabIndexProvider value={ lastFocusedElement }>
+			<div onFocus={ ( event ) => setLastFocusedElement( event.target ) } { ...props }>
```

Comment:
I would also consider putting this `onFocus` on `RovingTabIndexItem` instead. The way it is now, it would respond to elements that aren't wrapped within `RovingTabIndexItem`, which could lead to weird behaviors I guess. Not sure if it's a feature though! 😆

Reply:
That's a good point. I've moved the onFocus handler to the item. 👍